File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/GremlinWizard_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";


export class GremlinWizard_Model extends MonsterInfo{
    public static Id = "GremlinWizard"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(22, 26);
        else
            return Mathf.RandomRange(21, 25);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/wizardGremlin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/wizardGremlin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/wizardGremlin/skeleton.png"));
    }

    get Name(): string {
        return "地精法师";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/GremlinWizard_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class GremlinWizard_Model : MonsterInfo
    {
        public static string Id = "GremlinWizard";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(22, 26) : Random.Range(21, 25);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/wizardGremlin/skeleton.atlas",
                    "Monsters/theBottom/wizardGremlin/skeleton.json", "Monsters/theBottom/wizardGremlin/skeleton.png");
            }
        }

        public override string Name => "地精法师";

        public GremlinWizard_Model(Vector2 pos) : base(pos)
        {
        }
    }
}